src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/LocationSchedule.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity
+
+import jakarta.persistence.Entity
+import jakarta.persistence.EnumType
+import jakarta.persistence.Enumerated
+import jakarta.persistence.FetchType
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.Table
+import org.hibernate.Hibernate
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+@Entity
+@Table(name = "location_schedule")
+data class LocationSchedule(
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  val locationScheduleId: Long = 0,
+
+  @JoinColumn(name = "location_attribute_id")
+  @ManyToOne(fetch = FetchType.LAZY)
+  val locationAttribute: LocationAttribute,
+
+  val startDayOfWeek: Int,
+
+  val endDayOfWeek: Int,
+
+  val startTime: LocalTime,
+
+  val endTime: LocalTime,
+
+  @Enumerated(EnumType.STRING)
+  val locationUsage: LocationUsage,
+
+  val allowedParties: String? = null,
+
+  val notes: String? = null,
+
+  val createdBy: String,
+
+  val createdTime: LocalDateTime = LocalDateTime.now(),
+) {
+  var amendedBy: String? = null
```

Comment:
Above line 47 perhaps ...

```
init {
  require(locationAttribute.locationUsage == LocationUsage.SCHEDULE) {
    "blah blah" 
  }
}
```

Reply:
Just to explain my rational for putting validation in the Domain layer.

It means keeps the service layer functions simpler, less validation required in the service layer.

Reply:
Done.